bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/api/ApiHandler.java

```diff
@@ -0,0 +1,346 @@
+        }
+        return manager;
+    }
+
+    public synchronized @Nullable VmManager getVmManager() {
+        VmManager manager = (VmManager) managers.get(VmManager.class);
+        if (manager == null && getLoginManager().hasPermission(VmManager.associatedPermission())) {
+            manager = new VmManager(this);
+            managers.put(VmManager.class, manager);
+        }
+        return manager;
+    }
+
+    public synchronized @Nullable PlayerManager getPlayerManager() {
+        PlayerManager manager = (PlayerManager) managers.get(PlayerManager.class);
+        if (manager == null && getLoginManager().hasPermission(PlayerManager.associatedPermission())) {
+            manager = new PlayerManager(this, version);
+            managers.put(PlayerManager.class, manager);
+        }
+        return manager;
+    }
+
+    @SuppressWarnings("unchecked")
+    public <F, T extends Response<F>> F execute(URI url, HttpMethod method, @Nullable Object aPayload,
+            @Nullable Class<T> classOfT, boolean retryAuth) throws FreeboxException {
+        Object serialized = executeUrl(url, method, aPayload, classOfT, retryAuth, 3);
+        return classOfT != null ? ((T) serialized).getResult() : null;
+    }
+
+    <F, T extends ListResponse<F>> List<F> executeList(URI anUrl, HttpMethod method, @Nullable String aPayload,
+            Class<T> classOfT, boolean retryAuth) throws FreeboxException {
+        T serialized = executeUrl(anUrl, method, aPayload, classOfT, retryAuth, 3);
+        return serialized.getResult();
+    }
+
+    private String sendRequest(Request request) throws FreeboxException {
+        try {
+            ContentResponse response = request.timeout(DEFAULT_TIMEOUT_MS, TimeUnit.MILLISECONDS).send();
+            return new String(response.getContent(), StandardCharsets.UTF_8);
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            throw new FreeboxException("Exception while calling " + request.getURI(), e);
+        }
+    }
+
+    private synchronized <T extends BaseResponse> T executeUrl(URI url, HttpMethod method, @Nullable Object aPayload,
+            @Nullable Class<T> classOfT, boolean retryAuth, int retryCount) throws FreeboxException {
+        logger.debug("executeUrl {} - {} ", method, url);
+        try {
+            Request request = httpClient.newRequest(url).method(method);
+            httpHeaders.entrySet().forEach(entry -> request.header(entry.getKey(), entry.getValue()));
+
+            if (aPayload != null) {
+                String payload = gson.toJson(aPayload);
+                InputStream stream = new ByteArrayInputStream(payload.getBytes(StandardCharsets.UTF_8));
+                try (final InputStreamContentProvider contentProvider = new InputStreamContentProvider(stream)) {
+                    request.content(contentProvider, null);
+                }
+            }
+            T serialized = deserialize(classOfT, sendRequest(request));
```

Comment:
Please add a way to log API responses, like for example:
```
           String response = sendRequest(request);
            logger.trace("executeUrl {} - {} returned {}", method, url, response);
            T serialized = deserialize(classOfT, response
```
You should take to not log confidential data like I did in the original binding.